### PR TITLE
Make downloads respect IO

### LINKED ISSUE #2328: Artifact downloading does not respect io
```
$ JULIA_DEPOT_PATH=$(mktemp -d) julia --project=. -e 'using Pkg; Pkg.add("SQLite"; io=devnull)'
  Downloaded artifact: SQLite
```

### PR DESCRIPTION
**Pull request by IanButterworth**
I'm separating this out of #2355 given in that PR I also tried to fix the way download logging is done, which seems to need further thought.

Makes downloads respect the `io` kwarg
```
julia> Pkg.add("Images", io=devnull)

julia> 
```
Fixes #2328

### CODE REVIEW

**Comment by DilumAluthge** [comment]
Nice!

Is this the last part of Pkg that needs to be fixed to use `io`?

**Comment by IanButterworth** [comment]
> Is this the last part of Pkg that needs to be fixed to use `io`?

I just tried in a fresh depot, and the answer seems to be no
```
julia> Pkg.add("Images", io=devnull)
  Installing known registries into `/tmp/jdepot`
     Cloning registry from "https://github.com/JuliaRegistries/General.git"
       Added registry `General` to `/tmp/jdepot/registries/General`
```
Plus there were some ephemeral progress bars showing up

**Comment by IanButterworth** [comment]
Let's keep this to fixing #2328

Can someone approve or merge. Thanks

**Comment by fredrikekre** [comment]
`stdout` -> `stderr`?

**Comment by IanButterworth** [comment]
Perhaps the correct thing to do would be to make all of the `io` kwargs default to `DEFAULT_IO[]`, which defaults to `stderr` as the registry functions do?

https://github.com/JuliaLang/Pkg.jl/blob/47105aaefb7eb8bda5c002315ccf1a28be8fdabb/src/Registry/Registry.jl#L42

If people are monitoring `stdout` specifically for Pkg logging, that change would break things, but given `io` can be provided, it would arguably be the wrong way to do it anyway, so ok to break

**Comment by fredrikekre** [comment]
Yea sure, as long as it goes to stderr it is fine. (Already used here for downloads btw: https://github.com/JuliaLang/Pkg.jl/blob/f00fa969d96a50e96a5cd620b5b3dda7fd8ef12b/src/Artifacts.jl#L439)

File: src/API.jl [hunk 1]
```diff
@@ -1443,7 +1444,7 @@ end
 @deprecate status(mode::PackageMode) status(mode=mode)
 
 function status(ctx::Context, pkgs::Vector{PackageSpec}; diff::Bool=false, mode=PKGMODE_PROJECT,
-                io::IO=stdout, kwargs...)
```
Comment (fredrikekre) [inline src/API.jl:1446]: This should be `stdout` (this is output the user has asked for).

Comment (IanButterworth) [reply src/API.jl:1446]: or should it use `ctx.io` and remove the `io` kwarg?

Comment (IanButterworth) [reply src/API.jl:1446]: I get your point now.. I'll revert

**Comment by IanButterworth** [comment]
I think i've coalesced the Registry logging and the download progress meters that I was seeing to use `io` now.

At least in a fresh depot I get no output now, with this
```
julia> Pkg.add("VideoIO", io=devnull)

julia> Pkg.add("FFTW", io=devnull)

julia> 
```

**Pull request merged**